### DISCUSSION TITLE
Mochi-MQTT Organisation

### DISCUSSION THREAD

**Discussion by mochi-co** [mochi-mqtt/server#263] [General]
In light of the increasing popularity of this project, and my decreasing amount of available time to work on it, I am currently rolling around the idea of transferring the repo to a dedicated [mochi-mqtt organisation](http://github.com/mochi-mqtt). 

This move would also allow us to add more repos within the organisation, providing a centralised home for project like clustering, prometheus, and other community hooks. I'd still hope to make the occasional fix or feature, and push releases, and make time to help guide decision making/roadmapping and reviewing PRs. We've been lucky to have some excellent contributors in the past year, including those who use the project in production, and it may be time to migrate to a more group-centric approach. I feel confident at this stage that the project is valuable to many people, and it would be very exciting to see where it could go.

However, I would be interested to know what you think about this idea.

cc the most recent common contributors - @thedevop @dgduncan @bkupidura @wind-c @werbenhu @ianrose14

**Comment by dgduncan** [comment]
@mochi-co I agree with this move. Like you said, enough people use it that it is a bit onerous for you to bare the responsibility by yourself, especially if it continues to grow. With an organization as well, we may be able to better to come to a consensus on a light roadmap moving forward. I have a few personal things in the fire myself that are additions that could be useful. I would love to hear other's opinions.

**Comment by torntrousers** [comment]
Hello. I'm a heavy user of Mochi MQTT. Moving it to an organisation sounds like a positive thing, but its a slightly scary prospect to hear that you could be reducing the amount of time you spend on the project. Could you give any indication on how swift and dramatic that change to your involvement could be?

I've some experience with multi-group open source development, so if you'd like any help with setting up a governance structure for the organisation I'd be happy to contribute.

**Comment by bkupidura** [comment]
@mochi-co i totally agree with @torntrousers. For corporate/profesional use its better to have software managed by 'organization' and not single developer. But i believe you should stay core of it. Im afraid that if you step back now, project will slowly go into 'maintaned only' mode.
Whatever you decide, i will stay involved - as i think this is awesome piece of code.

**Comment by bddckr** [comment]
As a total newbie and outsider to the great work you all have been doing, it struck me and a colleague that this broker implementation is one of the more flexible ones thanks to the hook architecture. With that said, I'm not too worried that the main contributor/creator can contribute fewer hours to the project, as it seems like the project is in good shape - old enough for the kinks to be worked out by now, validated by the Paho test results when run against this broker.

As a potential user evaluating MQTT broker implementations, it would be great if the various forks could be "merged upstream". Seeing clustering available elsewhere is awesome, but it would certainly be easier to navigate as a new user if said feature is available under one shared project everyone keeps working on. It sounds like that's the intention of this move to a mochi organization. 🤞
> providing a centralised home for project like clustering, prometheus, and other community hooks

**Comment by mochi-co** [comment]
cc @torntrousers @bkupidura 
Just wanted to quickly update on this as I see there are some concerns. Rest assured, I haven't got any intention of 'releasing the reins' in the immediate or foreseeable future. It's very important to me that the codebase stays welcoming, maintainable, and most importantly, sensible (in honesty, my own code doesn't always work the first time, but this is why we have reviews). I also have plenty of ideas I'm still trying to figure out how to implement in a clean way (clustering, prometheus hooks... listeners as hooks... perhaps that's a step too far).

My motivation behind this discussion is more from a feeling that we can achieve more with a little democratisation. Besides which, becoming an organisation would provide some legitimacy and an opportunity to develop a roadmap, collaborate, and hopefully evolve the project into something truly beneficial. It also provides confidence for vendors, as you've said (this is the main argument against keeping it on my personal repo with collaborators). I've seen so many great projects where the main developer appears to vanish, and then we end up with a thousand forks and no collaboration - I feel that we all deserve better than that, and so I don't want it to be something people worry about.

On a purely personal and informal perspective, my current life circumstances and career keep me busier than I had maybe expected, and while I try to stay on top of issues and release, sometimes I'm not able to dedicate as much time as I would like - I don't want to be a roadblock which prevents the project from moving forward. Moving to an organisation would relieve some of that pressure for me.

But mostly I am just excited to have a better framework and opportunity to collaborate with good people.

**Reply by torntrousers** [reply]
Thanks for the comments, I agree with the sentiment and what you say is reassuring.

**Comment by mochi-co** [comment]
The first stage of this has been completed, see #271 for more details 👍🏻
